pkg/api/rest/request.go

```diff
@@ -0,0 +1,397 @@
+package rest
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"path"
+	"time"
+
+	terrors "github.com/pingcap/errors"
+	"github.com/pingcap/log"
+	"github.com/pingcap/ticdc/cdc/model"
+	cerrors "github.com/pingcap/ticdc/pkg/errors"
+	"github.com/pingcap/ticdc/pkg/httputil"
+	"github.com/pingcap/ticdc/pkg/retry"
+
+	"go.uber.org/zap"
+	"golang.org/x/time/rate"
+)
+
+const (
+	defaultBackoffBaseDelayInMs = 500
+	defaultBackoffMaxDelayInMs  = 30 * 1000
+	defaultMaxRetries           = 10
+)
+
+type HTTPClient interface {
```

Comment:
What is the motivation for this interface?

Reply:
Dead code is removed in new commit